client/tls_support.go

```diff
@@ -0,0 +1,162 @@
+package client
+
+import (
+	"fmt"
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/pkg/certs"
+	"github.com/skupperproject/skupper/pkg/kube"
+	"github.com/skupperproject/skupper/pkg/qdr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+	"time"
+)
+
+func (cli *VanClient) CreateServiceCA(ownerRef *metav1.OwnerReference) error {
+
+	ca := types.CertAuthority{Name: types.ServiceCaSecret}
+
+	caSecret, err := kube.NewCertAuthority(ca, ownerRef, cli.Namespace, cli.KubeClient)
+
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	_, err = kube.NewSimpleSecret(types.ServiceClientSecret, caSecret, ownerRef, cli.Namespace, cli.KubeClient)
+
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	err = cli.AppendSecretToRouter(types.ServiceClientSecret, false)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (cli *VanClient) CreateSecretForService(serviceName string, hosts string, secretName string) (*corev1.Secret, error) {
+	caCert, err := cli.KubeClient.CoreV1().Secrets(cli.Namespace).Get(types.ServiceCaSecret, metav1.GetOptions{})
+
+	if err != nil {
+		return nil, err
+	}
+
+	serviceSecret := certs.GenerateSecret(secretName, serviceName, hosts, caCert)
+	createdSecret, err := cli.KubeClient.CoreV1().Secrets(cli.Namespace).Create(&serviceSecret)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return createdSecret, nil
+}
+
+func (cli *VanClient) DeleteSecretForService(secretName string) error {
+	_, err := cli.KubeClient.CoreV1().Secrets(cli.Namespace).Get(secretName, metav1.GetOptions{})
+
+	if err == nil {
+
+		err = cli.KubeClient.CoreV1().Secrets(cli.Namespace).Delete(secretName, &metav1.DeleteOptions{})
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (cli *VanClient) AppendSecretToRouter(secretName string, waitForRestart bool) error {
```

Comment:
Suggest considering whether this is a general operation on the qdr config resource and/or an general operation on a deployment to append/remove volumes for a deployment. Possibly a combination of both.

Reply:
Thanks, this was removed from the client and moved into the qdr package, let me know if that makes sense to you.